### PR TITLE
Add ability to exclude directories in addition to files.

### PR DESCRIPTION
**Pull request by gurrinder**
Add ability to exclude directories in addition to files.
E.g. `exclude: ['Vendor/*']`
Related issue: https://github.com/realm/jazzy/issues/503

### CODE REVIEW

File: lib/jazzy/sourcekitten.rb [hunk 1]
```diff
@@ -409,7 +409,7 @@ def self.filter_excluded_files(json)
       excluded_files = Config.instance.excluded_files
       json.map do |doc|
         key = doc.keys.first
-        doc[key] unless excluded_files.include?(Pathname(key))
+        doc[key] unless excluded_files.detect{|file| key.include? file.to_s}
```
Comment (segiddins) [inline lib/jazzy/sourcekitten.rb]: Why not use File.fnmatch ?

Comment (gurrinder) [inline lib/jazzy/sourcekitten.rb]: no particular reason. Just didn't think of it :) I can make the change if you'd like.

**Comment by jpsim** [comment]
There are a few easy rubocop style violations that need to be addressed: https://travis-ci.org/realm/jazzy/builds/126147351

**Comment by gurrinder** [comment]
Thanks @jpsim . Updated.

**Comment by jpsim** [comment]
Thanks!

**Pull request merged**